### ISSUE TITLE
Diffrent point of view about class .CyclicCall() usage

### ISSUE THREAD

**Issue by GISED-Link** [benhar-dev/mobject-io#6]
First, thank you for this amazing work. This is THE thing I searched to get a decent coding guidelines. Just one thing:

https://github.com/benhar-dev/mobject-core/blob/3b642c1ad05c211ea7a5e8be90324150d8932271/docs/coding-guide.md?plain=1#L115-L117

Ok this make total sense for most of the things (I do) in OOP. **But I am against the CyclicCall method.**

For cyclic call I would prefer to use my FB not as 'pure' class but let them live in the real-time / cyclic context and thus use the body, the INPUT_VAR and OUTPUT_VAR section. Since we can call `SUPER^()` to call the implementation of the body of the parent class, I don't see why it should be avoided?

Imagine something like this:
```pascal
// Real time part, all fb are used as usual
gripper(sensorRP  := IN.I101_ST101_gripper_RP,
        sensorWP  := IN.I102_ST101_gripper_WP,
        out_valve => OUT.Q101_ST101_gripper);

// station step management
CASE stepCurrent OF
// ~ some code ~
ST101_Step_e.PICK_PIECE:

    gripper.Pick();

    // IF gripper.IsPicked() THEN or
    IF gripper.piecePicked THEN
        stepCurrent := ST101_Step_e.DO_SOMETHING;
    END_IF
// ~ some code ~
END_CASE
```

Isn't that beautiful? How could you do this using the cyclical call method? How the hardware input should be used by the function block? Of course a property is way more powerfull as a standard input, but do we need this power? Is there advantages of such a CyclicCall method?

**Comment by GISED-Link** [comment]
Ok, the more I read your documentation the more I understand where you will go with this project. This issue is irrelevant since you are thinking about to do something special with the CyclicCall method.

**Comment by benhar-dev** [comment]
Hi @GISED-Link, 

Thank you for your comments on here, it is much appreciated!

I totally agree with you with how nice code looks when you get in to OOP, and for many years I used a combination of FB style and CLASS in exactly the same way you describe.  With this style you will make your programs more readable and it makes code writing easier.  I coded this way for so long because it works.   

The problem I started to face is that I was applying OOP to more and more machines and projects, and as such I found that I was finding cyclic calling to be an issue.  In a machine it was not as obvious, but when I was coding projects which handled communication and request / response was writing myself in to a corner. 

Almost everything with IO needs cyclic calling, but also things such as timers.  If I passed a cyclic object in to another object which was non-cyclic then I shouldn't force the non-cyclic object to know or care to call the cyclic object.  This placed my cyclic call on the outside.  Still, totally fine and this can be done as most of the time your IO is static, so you know you need to call a motor or lamp.  

Then the big problem hit.  I wanted to use __New to make something which was cyclic.  What was worse was that the thing that created it was non-cyclic.   Hence Mobject was born.   Any objects can be cyclic, and can also automatically be called forever, or until they complete.  This allows me to make things like asynchronous commands, which only exist for a short period, but still need "updating" each cycle to do their job.  To be able to manage such a thing I needed to fully embrace OOP and remove the non-oop parts from my code.  

The examples on Mobject do not do a good job of conveying this problem/solution, as if you do not make things with __NEW then it's not obvious that this problem exsists.  FB style or CLASS with the call on the outside is totally fine.  As soon as you use __NEW you run head long in to the problem I had.  

I hope you like the work and I'm always keen to know peoples ideas and comments.   Plus I hope Mobject works for you as and when you need something cyclic called when you __NEW it up.  

Happy coding!

Ben - Coding bytes
https://codingbytes.teachable.com/p/codingbytes_twincat3

**Comment by GISED-Link** [comment]
Hi @benhar-dev,

Thank you very much for your reply. I open this issue waaaaay too early. After I open the issue, I started to read what was in the src folder. And only then I finally got the whole concept of this project.

You are clearly a step above where my team needs to go. Now we have duplicated code everywhere, early days code convention, etc...

Thank you for this amazing project, I hope I can give it a try in a next future.
